### PR TITLE
Define a package with which to support duck typed definitions.

### PR DESCRIPTION
**Pull request by mattmoor**
This also sketches out the skeletons for Targettable, Subscribable, and Conditions.

cc @vaikas-google WDYT?

### CODE REVIEW

**Comment by mattmoor** [comment]
cc @n3wscott

**Comment by vaikas-google** [comment]
Looks like a great start, need to figure out the deets, but I can't wait to try this out in Subscription.

**Comment by mattmoor** [comment]
Ok, I removed the hold.  Excited to see some of these get schemas and in particular blend with the Conditions work.

File: apis/duck/v1alpha1/init_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2017 The Knative Authors
```
Comment (n3wscott) [inline apis/duck/v1alpha1/init_test.go]: 2018

File: apis/duck/v1alpha1/register.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+var (
+	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	AddToScheme   = SchemeBuilder.AddToScheme
+)
+
+// Adds the list of known types to Scheme.
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(
+		SchemeGroupVersion,
+		// &VirtualService{},
```
Comment (n3wscott) [inline apis/duck/v1alpha1/register.go]: nit: clear this out

File: apis/duck/v1alpha1/subscribable_types.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+)
+
+// Subscribable is the schema for the subscribable portion of the payload
+type Subscribable struct {
+	// TODO(vaikas): Give me a schema!
+	Field string `json:"field,omitempty"`
+}
+
+// Implementations can verify that they implement Subscribable via:
+var _ = duck.VerifyType(&Topic{}, &Subscribable{})
```
Comment (n3wscott) [inline apis/duck/v1alpha1/subscribable_types.go]: nice!

**Pull request merged**